common/determined_common/api/authentication.py

```diff
@@ -164,7 +164,7 @@ def _write_store(self) -> None:
         self._create_det_path_if_necessary()
         cache_path = self._get_token_cache_path()
         store = {}
-        if self._tokens is not None and len(self._tokens):
```

Comment:
Honestly, I think because we use mypy, using on `if len(thing):` is superior to depending on the truthiness of `if thing:` because the truthiness test will sometimes hide bugs from mypy if you e.g. change `thing` to/from `Optional[ThingType]`.  So I'd prefer that these sorts of changes be omitted.

(also if we were going to depend on the truthiness test, the proposed edit is redundant anyway)